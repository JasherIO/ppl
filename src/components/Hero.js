import React from 'react'
import { Link, graphql, StaticQuery } from 'gatsby'
import Img from 'gatsby-image'

export const PureSection = ({ data }) => {
  const hero = data.markdownRemark

  return (
    <div className="section container gradient" style={{ 
      padding: 0
    }}>
      <Link to={hero.fields.slug}>
        <figure className="image">
          <Img fluid={hero.frontmatter.cover.childImageSharp.fluid} style={{borderRadius: "10px"}} />
        </figure>
      </Link>

      <div className="content" style={{
        position: "absolute",
        bottom: "0",
        left: "1rem"
      }}>
        <h1 className="title is-2 has-text-white">{hero.frontmatter.title}</h1>
      </div>
    </div>
  )
}

const query = graphql`
  query {
    markdownRemark(frontmatter: {templateKey: {in: ["post"]}, feature: {eq: true}}) {
      fields {
        slug
      }
      frontmatter {
        title
        cover {
          childImageSharp {
            fluid(maxWidth: 1500) {
              ...GatsbyImageSharpFluid_withWebp
            }
          }
        }
      }
    }
  }
`

export const Section = props => (
  <StaticQuery query={query} render={data => <PureSection data={data} {...props} />} />
)

export default Section