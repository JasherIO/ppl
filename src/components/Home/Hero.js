import React from 'react'
import { graphql, StaticQuery } from 'gatsby'

const backgroundImage = (img) => {
  return `linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url(${img})`
}

const Section = ({ data }) => {
  const { hero } = data.markdownRemark.frontmatter
  
  return (
    <section className="hero is-primary is-medium hero-background" style={{ backgroundImage: backgroundImage(hero.img) }}>
      <div className="hero-body">
        <div className="container">
          <div className="title">
            {hero.title}
          </div>
          <div className="subtitle">
            {hero.subtitle}
          </div>
        </div>
      </div>
    </section>
  )
}

const query = graphql`
  query {
    markdownRemark(frontmatter: { templateKey: { eq: "home-post" } }) {
      frontmatter {
        hero {
          title
          subtitle
          img
        }
      }
    }
  }
`

export default props => (
  <StaticQuery query={query} render={data => <Section data={data} {...props} />} />
)