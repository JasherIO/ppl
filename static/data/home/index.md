---
templateKey: home-post
tabs:
  - name: Players
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus facilisis, nisl quis porta bibendum, nulla purus vestibulum sem, sit amet ultrices nisi elit ut odio. Sed rutrum eu velit at rhoncus.'
  - name: Teams
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus facilisis, nisl quis porta bibendum, nulla purus vestibulum sem, sit amet ultrices nisi elit ut odio. Sed rutrum eu velit at rhoncus.'
  - name: Format
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus facilisis, nisl quis porta bibendum, nulla purus vestibulum sem, sit amet ultrices nisi elit ut odio. Sed rutrum eu velit at rhoncus.'
sponsors:
  - img: /img/AllYours.png
    link: 'https://google.com'
    name: All Yours Buffet
  - img: /img/WhatASave.png
    link: 'https://google.com'
    name: What a Save Esports
  - img: /img/EggingGrocers.png
    link: 'https://google.com'
    name: Egging Grocers
---

