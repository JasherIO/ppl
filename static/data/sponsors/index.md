---
templateKey: sponsor-post
sponsors:
  - img: /img/AllYours.png
    link: 'https://google.com'
    name: All Yours Buffet
  - img: /img/WhatASave.png
    link: 'https://google.com'
    name: What a Save Esports
  - img: /img/EggingGrocers.png
    link: 'https://google.com'
    name: Egging Grocers
---

