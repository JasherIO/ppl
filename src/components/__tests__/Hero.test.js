import React from "react"
import renderer from "react-test-renderer"

import { PureSection } from "../Hero"
import data from './__data__/home.json'

describe("Hero Section", () =>
  it("renders correctly", () => {
    const tree = renderer
      .create(<PureSection data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))