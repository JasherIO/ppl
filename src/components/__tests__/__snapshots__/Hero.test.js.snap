// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Section renders correctly 1`] = `
<section
  className="hero is-primary is-medium hero-background"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url(/img/Shiawese.jpg)",
    }
  }
>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        Welcome to the official site for Pulsar Premier League!
      </h1>
      <h2
        className="subtitle"
      >
        Notice: The site is unfinished and information posted is for demo purposes. 
      </h2>
    </div>
  </div>
</section>
`;
